docs/howto/custom-template-tags.txt

```diff
@@ -364,6 +364,28 @@ Template filter code falls into one of two situations:
 
 .. _filters-timezones:
 
+Auto-escaping with a decorator
+-------------------------------
+
+Using the :meth:`django.template.defaultfilters.autoescape_aware` decorator, filters can be made to
+automatically escape text based on the value of the autoescape parameter.
+If the filter's ``autoescape`` parameter is ``True``, the first argument
+will be escaped. Otherwise, it will be left as is.
+
+This is the equivalent of adding the following at the beginning of the filter:
+
+.. code-block:: python
+
+    if autoescape:
+        esc = conditional_escape
+    else:
+        esc = lambda x: x
+    text = esc(text)
```

Comment:
This is the code which features in `Filters and Auto-escaping, point 2` above. It's that that the new decorator here is meant to replace (or provide an option for). 

As such, I think this block, and usage instructions, need to be worked into the text there, rather than bolted on at the end here. 

So underneath the example there, I'd add a block saying _"This pattern of using conditional_escape if autoescape is enabled is common enough that there is a decorator that you can use..."